Cargo.toml

```diff
@@ -196,161 +196,161 @@ rust-version = "1.83.0"
-mc-oblivious-traits = "2.3"
-mc-rand = "1.1.0"
-mc-sgx-core-sys-types = "0.12.0"
-mc-sgx-core-types = "0.12.0"
-mc-sgx-dcap-ql = "0.12.0"
-mc-sgx-dcap-quoteverify = "0.12.0"
-mc-sgx-dcap-sys-types = { version = "0.12.0", default-features = false }
-mc-sgx-dcap-types = { version = "0.12.0", default-features = false }
-merlin = { version = "3.0", default-features = false }
-mockall = "0.12.1"
+libz-sys = "1"
+link-cplusplus = "1"
+lmdb-rkv = "0.14"
+log = "0.4"
+maplit = "1"
+mbedtls = { version = "0.8", default-features = false }
+mbedtls-sys-auto = "2"
+mc-attestation-verifier = "0.4"
+mc-oblivious-aes-gcm = { version = "0.10", default-features = false }
+mc-oblivious-map = "2"
+mc-oblivious-ram = "2"
+mc-oblivious-traits = "2"
+mc-rand = "1"
+mc-sgx-core-sys-types = "0.12"
+mc-sgx-core-types = "0.12"
+mc-sgx-dcap-ql = "0.12"
+mc-sgx-dcap-quoteverify = "0.12"
+mc-sgx-dcap-sys-types = { version = "0.12", default-features = false }
+mc-sgx-dcap-types = { version = "0.12", default-features = false }
+merlin = { version = "3", default-features = false }
+mockall = "0.12"
 more-asserts = "0.3"
-num_cpus = "1.16"
-once_cell = { version = "1.19", default-features = false }
-opentelemetry = "0.21.0"
-opentelemetry-jaeger = "0.20.0"
-opentelemetry_sdk = "0.21.2"
-p256 = { version = "0.13.0", default-features = false }
-pem = "3.0"
-percent-encoding = "2.3.1"
+num_cpus = "1"
+once_cell = { version = "1", default-features = false }
+opentelemetry = "0.21"
+opentelemetry-jaeger = "0.20"
+opentelemetry_sdk = "0.21"
+p256 = { version = "0.13", default-features = false }
+pem = "3"
+percent-encoding = "2"
 pkg-config = "0.3"
-portpicker = "0.1.1"
+portpicker = "0.1"
 predicates = "3"
-primitive-types = "0.12.2"
-proc-macro2 = "1.0.8"
-prometheus = "0.13.3"
-proptest = { version = "1.4", default-features = false }
+primitive-types = "0.12"
+proc-macro2 = "1"
+prometheus = "0.13"
+proptest = { version = "1", default-features = false }
 prost = { version = "0.12", default-features = false }
 prost-build = "0.12"
-protobuf = "2.27.1"
-protoc-grpcio = "3.0.0"
-quote = "1.0.2"
-r2d2 = "0.8.10"
-rand = { version = "0.8.5", default-features = false }
+protobuf = "2"
+protoc-grpcio = "3"
+quote = "1"
+r2d2 = "0.8"
+rand = { version = "0.8", default-features = false }
 rand_chacha = "0.3"
-rand_core = { version = "0.6.4", default-features = false }
+rand_core = { version = "0.6", default-features = false }
 rand_hc = { version = "0.3", default-features = false }
-rayon = "1.9"
+rayon = "1"
 regex = "1"
 reqwest = { version = "0.11", default-features = false }
-retry = "2.0"
-rjson = "0.3.1"
-rocket = "0.5.1"
-rusoto_core = { version = "0.48.0", default-features = false }
-rusoto_s3 = { version = "0.48.0", default-features = false }
+retry = "2"
+rjson = "0.3"
+rocket = "0.5"
+rusoto_core = { version = "0.48", default-features = false }
+rusoto_s3 = { version = "0.48", default-features = false }
 schnorrkel-og = { version = "0.11.0-pre.0", default-features = false }
```

Comment:
`-pre` versions receive special treatment and cargo requires that they are explicitly specified. This dependency is overridden in the `[patch.crates-io]` section at the bottom to point to a specific commit of this specific version.